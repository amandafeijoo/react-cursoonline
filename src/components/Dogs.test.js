import { render, screen } from "@testing-library/react"
import Dog from "./Dogs"

test('Should render without errors', ()=>{
render(<Dog/>)
})
//se tiene que llamar a render y aqui hay que pasarle el componente.


test('should render a dog name', () => {
    render (<Dog name= "India" />)
    expect(screen.getByText("India")).toBeInTheDocument()  
})

test('should render a dog age name', () => {
    render (<Dog name= "India" age={7} />)
    expect(screen.getByText("India")).toBeInTheDocument()  
})

test('should render a image', () => {
    render (<Dog name= "India" age={7} img= "https://salchichascolombia.com/wp-content/uploads/2022/05/IMG_5542-1-793x1024.jpeg" />)
    expect(screen.getByRole("img").src).toBe("https://salchichascolombia.com/wp-content/uploads/2022/05/IMG_5542-1-793x1024.jpeg")  
})

test('should render a children text', () => {
    render (<Dog name= "India" age={7} img= "https://salchichascolombia.com/wp-content/uploads/2022/05/IMG_5542-1-793x1024.jpeg" >test</Dog>)
    expect(screen.getByText("test")).toBeInTheDocument()
})


// Screen sirve para buscar elementos no los importa del testing library 