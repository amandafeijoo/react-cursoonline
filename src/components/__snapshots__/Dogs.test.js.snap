// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div>
  <div>
    <img
      height="300px"
    />
    <p>
       
       
    </p>
    <p>
       
    </p>
    <p>
       
       
    </p>
  </div>
</div>
`;
